unittests/allplatformstests.py

```diff
@@ -1547,47 +1549,61 @@ def dist_impl(self, vcs_init, vcs_add_all=None, include_subprojects=True):
-                                 'disttest-1.4.3/subprojects/samerepo/meson.build']
-                self.assertEqual(sorted(expected),
-                                 sorted(z.namelist()))
-                # Verify that with --include-subprojects we now also have files
-                # from tarball and separate vcs subprojects. But not files from
-                # unused subprojects.
-                self._run(self.meson_command + ['dist', '--formats', 'zip', '--include-subprojects'],
-                          workdir=self.builddir)
-                z = zipfile.ZipFile(zip_distfile)
-                expected += ['disttest-1.4.3/subprojects/tarballsub/',
-                             'disttest-1.4.3/subprojects/tarballsub/meson.build',
-                             'disttest-1.4.3/subprojects/vcssub/',
-                             'disttest-1.4.3/subprojects/vcssub/meson.build']
-                self.assertEqual(sorted(expected),
-                                 sorted(z.namelist()))
-            if vcs_add_all:
-                # Verify we can distribute separately subprojects in the same vcs
-                # repository as the main project.
-                subproject_dir = os.path.join(project_dir, 'subprojects', 'samerepo')
-                self.new_builddir()
-                self.init(subproject_dir)
-                self.build('dist')
-                xz_distfile = os.path.join(self.distdir, 'samerepo-1.0.tar.xz')
-                xz_checksumfile = xz_distfile + '.sha256sum'
-                self.assertPathExists(xz_distfile)
-                self.assertPathExists(xz_checksumfile)
-                tar = tarfile.open(xz_distfile, "r:xz")  # [ignore encoding]
-                self.assertEqual(sorted(['samerepo-1.0',
-                                         'samerepo-1.0/meson.build']),
-                                 sorted(i.name for i in tar))
+                            'disttest-1.4.3/distexe.c',
+                            'disttest-1.4.3/subprojects/',
+                            'disttest-1.4.3/subprojects/samerepo/',
+                            'disttest-1.4.3/subprojects/samerepo/meson.build'])
+
+            obtained = set(z.namelist())
+            if vcs_init.__name__ == "hg_init":
+                obtained.remove("disttest-1.4.3/.hg_archival.txt")
+
+            self.assertEqual(expected, obtained)
+            # Verify that with --include-subprojects we now also have files
+            # from tarball and separate vcs subprojects. But not files from
+            # unused subprojects.
+            self._run(self.meson_command + ['dist', '--formats', 'zip', '--include-subprojects'],
+                        workdir=self.builddir)
+            z = zipfile.ZipFile(zip_distfile)
+
+            obtained = set(z.namelist())
+            if vcs_init.__name__ == "hg_init":
+                obtained = {item for item in obtained
+                            if not item.endswith(".hg_archival.txt")}
```

Comment:
Is the inclusion of the `.hg_archival.txt` metadata file a behavior change? Should the `-X ui.archivemeta=false` option be passed to `hg archive`?